app/models/template.rb

```diff
@@ -11,8 +11,8 @@ class Template < ApplicationRecord
   validates :name, :presence => true
   validates :template, :presence => true
   validates :audit_comment, :length => {:maximum => 255}
-  validate :template_changes, :if => ->(template) { (template.locked? || template.locked_changed?) && template.persisted? && !Foreman.in_rake? }
-  validate :inputs_unchanged_when_locked, :if => ->(template) { (template.locked? || template.locked_changed?) && template.persisted? && !Foreman.in_rake? }
+  validate :template_changes, :if => ->(template) { run_template_changes_validation?(template) }
```

Comment:
You can use the function directly, no need to go through a lambda:
```suggestion
  validate :template_changes, if: :run_template_changes_validation?
```
see https://guides.rubyonrails.org/active_record_validations.html#conditional-validation for more info on validations